whois-query/src/main/java/net/ripe/db/whois/query/pipeline/ConnectionStateHandler.java

```diff
@@ -39,24 +37,25 @@ public void messageReceived(final ChannelHandlerContext ctx, final MessageEvent
         }
 
         if (query.hasOnlyKeepAlive()) {
-            channel.getPipeline().sendDownstream(new QueryCompletedEvent(channel));
+            channel.pipeline().write(new QueryCompletedEvent(channel));
```

Comment:
As discussed at standup, review write() vs. writeAndFlush()

Reply:
I looked a bit into this and calling `writeAndFlush` always might not be the way to go. Based on the comment [here](https://github.com/netty/netty/issues/6609#issuecomment-292205575) and [here](http://normanmaurer.me/presentations/2014-facebook-eng-netty/slides.html#6.0). Everything will still works as long as the last pipeline has the `writeAndFlush` call.

I'll leave it as is now, and within the client-ip changes will look at the possibility of perhaps optimising the use of `write` vs `writeAndFlush`